WIP: V1

luebken opened the pull request:
My WIP branch to allow comments and discussion for a first version.

luebken committed:
first resources

luebken committed:
outline

luebken committed:
A first version on "Proper Linux Process"

luebken committed:
some formatting and reworing of Proper Linux process

luebken committed:
first draft explicit interfaces

luebken committed:
splitting and content for "explicit interfaces"

luebken committed:
disposable

luebken committed:
content from slides

luebken committed:
redhat blogpost

luebken committed:
small additions

luebken committed:
links to image guidelines

luebken committed:
todos

Commit:
renamed file

Commit:
self-contained

Commit:
disposable

Commit:
added container-api

luebken committed:
resource video

luebken committed:
renamed to resources.md

luebken committed:
a first readme

grdryn committed:
Fixes for minor spelling and grammar issues

luebken committed:
some wording

luebken committed:
merged

luebken committed:
Merge pull request #5 from luebken/grdryn-minor-typo-fixes

Grdryn minor typo fixes

luebken committed:
mentioning dies

luebken committed:
fix link

luebken committed:
moved todo into issue

luebken committed:
cleanup todos in API

luebken committed:
cleanup of labels

luebken committed:
cleanup disposable

luebken committed:
immutability

luebken committed:
little rewording

luebken committed:
rewording and reorganising

luebken committed:
fixes

luebken committed:
merged

luebken committed:
dummy document for composite patterns

luebken committed:
some rewording

luebken committed:
rewording

luebken committed:
more wording

luebken committed:
some fine tuning for linux process

luebken committed:
cleanup in API

luebken committed:
intro to descriptive

luebken committed:
format

luebken committed:
WIP label

luebken committed:
fedora patch

luebken committed:
reasons for using multiple containers

luebken committed:
added health checks

luebken committed:
more reasons

zeisss commented on module-container.md:
```diff
@@ -0,0 +1,312 @@
+
+Before we come up with to many new ideas we should acknowledge the fact that a container is foremost a Linux process. Therefore we should apply common standards and best practices for writing Unix tools which happen to be containers. These are _React to signals_, _Use standard streams_, _Handle arguments_:
+
+#### React to signals
```
You probably at least mention the whole pid1 problematic (or link to where is discussed). If the root process inside a (docker) container spawns child processes and they die, the parent or root gets a signal to clean that up. If this process doesn't handle that signal, the process is kept in a zombie status.

zeisss commented on module-container.md:
````diff
@@ -0,0 +1,312 @@
+``` 
+https://github.com/luebken/currentweather/blob/master/server.js#L49
+
+**Docker best practices:**
````
I would actually not call this best practice but just something to be aware of. If you do it wrong but try to build a good container it may take a long time to find this.

luebken replied:
Hmmm. But the scheduler / user is sending signals to the container. I would think the container should react to it. Eg. to make it faster or to clean up.

In terms of wrong signal handling: Sure you have to do it right. But IMHO this is not an argument against it.

zeisss replied:
ack :)

zeisss commented on module-container.md:
```diff
@@ -0,0 +1,312 @@
+
+**General best practices:**
+
+* If you do have specific logging requirements you can use a side-car container to adapt your logs. See the composite patterns for details.
```
add a link to the other document here.

zeisss commented on module-container.md:
````diff
@@ -0,0 +1,312 @@
+* If your app writes to a file you could link that file to the device file:
+
+  ```Dockerfile
+  RUN ln -sf /dev/stdout /var/log/nginx/access.log
````
nice idea :)

zeisss commented on module-container.md:
```diff
@@ -0,0 +1,312 @@
+
+  * Faster in the network (deploy, reschedule, update)
+  * Increased I/O performance
+  * Smaller attack service. Easier to audit.
```
s/service/surface/

zeisss commented:
nice :)

luebken committed:
spelling

luebken committed:
link to pid one problem

luebken committed:
link

luebken committed:
suggestions for readme by @rosscdh
https://github.com/luebken/container-patterns/pull/11/commits/1dc570ae3b7e96a012342138ab105dcdc1f5ec23

luebken committed:
cherrpick wording suggestions for desc and definition by @rosscdh https://github.com/luebken/container-patterns/pull/11/commits/59e84f838871a96f88696e4e54f1f5bfa087bef9

luebken committed:
wording suggestions for linux process by @rosscdh https://github.com/luebken/container-patterns/pull/11/commits/59e84f838871a96f88696e4e54f1f5bfa087bef9

luebken committed:
cherry pick wording by @rosscdh and further clarification

luebken committed:
descriptive & disposable

luebken committed:
cherry picked wording suggestions by @rosscdh https://github.com/luebken/container-patterns/pull/11/commits/59e84f838871a96f88696e4e54f1f5bfa087bef9

luebken committed:
contributors

luebken committed:
format

luebken committed:
added link to gettext envsubst

luebken committed:
reference secrets

luebken committed:
link to autopilot

luebken committed:
Added orchestration agnostic.

luebken committed:
todo

luebken committed:
todo

luebken committed:
minor rewording

luebken committed:
Renamed Explicit interfaces to Self-Describing

luebken committed:
removed explicit interface

luebken committed:
merged master
